pkcs11/_pkcs11.pyx

```diff
@@ -480,68 +595,121 @@ class Token(types.Token):
             pin_length = <CK_ULONG> len(pin)
 
             with nogil:
-                retval = _funclist.C_Login(handle, final_user_type, pin_data, pin_length)
+                retval = self.funclist.C_Login(handle, c_user_type, pin_data, pin_length)
             assertRV(retval)
 
-        return Session(self, handle, rw=rw, user_type=user_type)
+        return Session.make(self, handle, rw=<bint> rw, user_type=c_user_type)
 
+    def __str__(self):
+        return self.label
 
-class SearchIter:
-    """Iterate a search for objects on a session."""
+    def _identity(self):
+        return Token.__name__, self.slot
 
-    def __init__(self, session, attrs):
+    def __repr__(self):
+        return "<{klass} (label='{label}' serial={serial} flags={flags})>".format(
+            klass=type(self).__name__, label=self.label, serial=self.serial, flags=str(self.flags)
+        )
+
+
+cdef class OperationContext:
```

Comment:
This is the main context manager class that monitors operations. There are subtypes for search & crypto ops.